packages/db-base/lib/inMemFileDB.js

```diff
@@ -349,7 +349,7 @@ class InMemoryFileDB {
     }
 
     // Destructor of the class. Called by shutting down.
-    destroy() {
+    async destroy() {
```

Comment:
That seems unnecessary. You're not `await`ing.

Reply:
If `this.saveState()` returns a Promise you either need to await it or just return the Promise.

Reply:
But a method dignature should be identical when extending or !? So if I change it to "async destroy" in a child class then I also should have it defined that waa y in parent class ?

Reply:
saveState is sync

Reply:
Yes and no. If you expect the derived methods to be async, this should at least have a return type of `Promise<void>`. Easiest way to do that is the async annotation like you did.

Reply:
I do not get your point

Reply:
I was agreeing with you ;) If the method is not intended to return something but child classes should return a Promise, either
```js
/**
 * ...
 * @returns {Promise<void>}
 */
destroy() {...}
```
or
```js
async destroy() {...}
```
do the trick.

The first is your only option if you also want to enforce that the Promise resolves to a number.